src/Features/Core/Portable/MoveToNamespace/AbstractMoveToNamespaceCodeAction.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.MoveToNamespace
+{
+    internal abstract partial class AbstractMoveToNamespaceCodeAction : CodeActionWithOptions
+    {
+        private readonly AbstractMoveToNamespaceService _changeNamespaceService;
+        private readonly MoveToNamespaceAnalysisResult _moveToNamespaceAnalysisResult;
+
+        public AbstractMoveToNamespaceCodeAction(AbstractMoveToNamespaceService changeNamespaceService, MoveToNamespaceAnalysisResult analysisResult)
+        {
+            _changeNamespaceService = changeNamespaceService;
+            _moveToNamespaceAnalysisResult = analysisResult;
+        }
+
+        public override object GetOptions(CancellationToken cancellationToken)
+        {
+            return _changeNamespaceService.GetOptionsAsync(
```

Comment:
Consider `GetChangeNamespaceOptionsAsync` so it's clear you're not talking about workspace options or any other kind of options.